### PR TITLE
quast 4.3 and DOI update

### PR DESCRIPTION
**Pull request by alexeigurevich**
### Have you:
- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?
### Updates to existing formula: have you
- [x] Removed the `revision` line, if any, when bumping the version number?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

### CODE REVIEW

**Comment by sjackman** [comment]
The CI test failed on all versions of Mac OS.

See https://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/5933/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/install_homebrew_science_quast/

```
==> /usr/local/Cellar/quast/4.3/bin/quast --test
/usr/local/Cellar/quast/4.3/quast.py --test

Version: 4.3, 87c74f5

System information:
  OS: Darwin-15.6.0-x86_64-i386-64bit (macosx)
  Python version: 2.7.10
  CPUs number: 4

Started: 2016-09-26 19:23:40

Logging to /private/tmp/quast-20160926-98119-15jx4i9/quast-4.3/quast_test_output/quast.log
NOTICE: Maximum number of threads is set to 1 (use --threads option to set it manually)

CWD: /private/tmp/quast-20160926-98119-15jx4i9/quast-4.3
Main parameters: 
  Threads: 1, eukaryotic: true, minimum contig length: 500, ambiguity: one, threshold for extensive misassembly size: 1000

Reference:
  /usr/local/Cellar/quast/4.3/test_data/reference.fasta.gz ==> reference

Contigs:
  Pre-processing...
  1  /usr/local/Cellar/quast/4.3/test_data/contigs_1.fasta ==> contigs_1
  2  /usr/local/Cellar/quast/4.3/test_data/contigs_2.fasta ==> contigs_2

2016-09-26 19:23:40
Running GAGE...
Compiling JAVA classes (details are in /usr/local/Cellar/quast/4.3/quast_libs/gage/make.log and make.err)
  1  contigs_1...
  1  Logging to files gage_contigs_1.stdout and gage_contigs_1.stderr...
  1  sh /usr/local/Cellar/quast/4.3/quast_libs/gage/getCorrectnessStats.sh /usr/local/Cellar/quast/4.3/quast_libs/E-MEM-osx \
     quast_test_output/quast_corrected_input/reference.fasta quast_test_output/quast_corrected_input/contigs_1.fasta \
     quast_test_output/gage/tmp 500 > quast_test_output/gage/gage_contigs_1.stdout \
     2> quast_test_output/gage/gage_contigs_1.stderr
  1  Failed.
  2  contigs_2...
  2  Logging to files gage_contigs_2.stdout and gage_contigs_2.stderr...
  2  sh /usr/local/Cellar/quast/4.3/quast_libs/gage/getCorrectnessStats.sh /usr/local/Cellar/quast/4.3/quast_libs/E-MEM-osx \
     quast_test_output/quast_corrected_input/reference.fasta quast_test_output/quast_corrected_input/contigs_2.fasta \
     quast_test_output/gage/tmp 500 > quast_test_output/gage/gage_contigs_2.stdout \
     2> quast_test_output/gage/gage_contigs_2.stderr
  2  Failed.
WARNING: Error occurred while GAGE was processing assemblies. See GAGE error logs for details: /private/tmp/quast-20160926-98119-15jx4i9/quast-4.3/quast_test_output/gage/gage_*.stderr
```

**Comment by alexeigurevich** [comment]
Hi! We found the problem. We included precompiled E-MEM aligner in quast-4.3, but it was compiled using gcc instead of clang (E-MEM cannot be compiled with clang). This caused the crashes on test Mac machines without gcc installed.
Currently, we are preparing a new release (4.4) which will include a fix for this issue (just a simple switch to regular Nucmer aligner if anything is bad with E-MEM). I hope the release will be ready within few weeks.
To sum up, I think we should wait and make a homebrew update from 4.2 to 4.4 directly.

**Comment by sjackman** [comment]
Would you be able to submit a separate formula for `E-MEM`, and instead have QUAST depend on `e-mem`? That'd be the tidier fix I think, rather than including a precompiled executable of E-MEM for every platform and OS.

**Comment by sjackman** [comment]
> just a simple switch to regular Nucmer aligner if anything is bad with E-MEM

If Nucmer gives different results than E-MEM, then automagically falling back to Nucmer is dangerous. I'd suggest that whether to use E-MEM or Nucmer should be an explicit command line option.

**Comment by alexeigurevich** [comment]
E-MEM and Nucmer produce absolutely the same results but E-MEM is faster, that is why we set it by default. However, it has more dependencies, so we have tricky stuff to make it work on different platforms (e.g. precompiled version for Mac, sources for Linux) or at least detect that it does not work and switch to Nucmer.
We fixed MacOS issue in 4.4: our v4.3 check system was not able to detect that E-MEM is failing there if gcc is not installed. The check should work correctly in other environments, too. On the first quast.py --test run we check whether E-MEM was able to align something or not and switch default aligner to Nucmer if needed.

Sorry for late 4.4 release, we expected to complete it much earlier but were delayed by some unexpected problems & feature requests. I hope it will be ready really soon.

**Comment by sjackman** [comment]
> E-MEM and Nucmer produce absolutely the same results but E-MEM is faster

Fantastic. I didn't realize that.

**Comment by sjackman** [comment]
> e.g. precompiled version for Mac, sources for Linux

Can you provide an option to use a preinstalled copy of E-MEM, installed for example by Linuxbrew? I prefer that to bundling dependencies.

**Comment by sjackman** [comment]
> Sorry for late 4.4 release, we expected to complete it much earlier but were delayed by some unexpected problems & feature requests. I hope it will be ready really soon.

No worries. I look forward to 4.4. Thanks for all your work on QUAST. It's a very useful tool.

**Comment by sjackman** [comment]
I've created a formula for E-MEM at https://github.com/Homebrew/homebrew-science/pull/4635

**Comment by alexeigurevich** [comment]
Great, thanks!
What do you think, if we add lookup for preinstalled E-MEM in PATH environment variable would it work? Or do we still need a separate option for specifying path to E-MEM installation?
Currently we have similar case with blast binaries which are needed for MetaQUAST without references mode. First, we check PATH and if blast is already installed on the system we use this installation, otherwise we download our own blast binaries to Quast installation folder and use them.

**Comment by sjackman** [comment]
Yep, checking for `e-mem` in `PATH` makes sense to me.

**Comment by alexeigurevich** [comment]
We finally released 4.4, so I've created a new pull-request here:
https://github.com/Homebrew/homebrew-science/pull/4757

**Comment by sjackman** [comment]
Superseded by #4757
